src-plugins/vtkDataMesh/readers/vtkDataMeshReader.cpp

```diff
@@ -86,140 +85,110 @@ bool vtkDataMeshReader::read(const QString& path) {
         }
         else
         {
-            qDebug() << "Loading the vtkDataMesh failed, it's neither a surface or volume mesh !";
+            qDebug() << metaObject()->className() << ": loading " << path << " failed, not a surface or volume mesh";
             return false;
         }
 
-        // Get the extension of the filename
-        QFileInfo pathfile(path);
-        QString extension = pathfile.completeSuffix();
-
-        if (extension == "vtk") // VTK files
-        {
-            // Extract data and header from the file
-            QString header;
-            vtkSmartPointer<vtkPolyDataReader> reader = vtkPolyDataReader::New();
-            try
-            {
-                reader->SetFileName(path.toLocal8Bit().constData());
-                reader->Update();
-
-                header = reader->GetHeader();
-                dataSet->Read(path.toLocal8Bit().constData());
-            }
-            catch (vtkErrorCode::ErrorIds error)
-            {
-                qDebug() << "vtkDataMeshReader: " << vtkErrorCode::GetStringFromErrorCode(error);
-                return false;
-            }
-
-            medData->setData(dataSet);
-
-            // Parse header and save metadata in medData
-            parseHeaderVtk(header, medData);
-        }
-        else if (extension == "vtp") // VTP files
+        try
```

Comment:
I think it's better to have all the reading code in the try block, so we don't need to care about which code in the reading part will actually throw an exception.